File: app/src/main/java/com/waynebloom/rotmgdpscalculator/StatBonus.java
package com.waynebloom.rotmgdpscalculator;

class StatBonus {
    int attBonus;
    int defBonus;
    int spdBonus;
    int dexBonus;
    int vitBonus;
    int wisBonus;
    int lifeBonus;
    int manaBonus;

    StatBonus() {

    }

    StatBonus(int attBonus, int defBonus, int spdBonus, int dexBonus, int wisBonus, int vitBonus, int lifeBonus, int manaBonus) {
        this.attBonus = attBonus;
        this.defBonus = defBonus;
        this.spdBonus = spdBonus;
        this.dexBonus = dexBonus;
        this.wisBonus = wisBonus;
        this.vitBonus = vitBonus;
        this.lifeBonus = lifeBonus;
        this.manaBonus = manaBonus;
    }

    public void addBonus(StatBonus statBonus) {
        attBonus = attBonus + statBonus.getAttBonus();
        defBonus = defBonus + statBonus.getDefBonus();
        spdBonus = spdBonus + statBonus.getSpdBonus();
        dexBonus = dexBonus + statBonus.getDexBonus();
        wisBonus = wisBonus + statBonus.getWisBonus();
        vitBonus = vitBonus + statBonus.getVitBonus();
        lifeBonus = lifeBonus + statBonus.getLifeBonus();
        manaBonus = manaBonus + statBonus.getManaBonus();
    }

    public void subtractBonus(StatBonus statBonus) {
        attBonus = attBonus - statBonus.getAttBonus();
        defBonus = defBonus - statBonus.getDefBonus();
        spdBonus = spdBonus - statBonus.getSpdBonus();
        dexBonus = dexBonus - statBonus.getDexBonus();
        wisBonus = wisBonus - statBonus.getWisBonus();
        vitBonus = vitBonus - statBonus.getVitBonus();
        lifeBonus = lifeBonus - statBonus.getLifeBonus();
        manaBonus = manaBonus - statBonus.getManaBonus();
    }

    public void replaceBonus(StatBonus statBonus) {
        attBonus = statBonus.getAttBonus();
        defBonus = statBonus.getDefBonus();
        spdBonus = statBonus.getSpdBonus();
        dexBonus = statBonus.getDexBonus();
        wisBonus = statBonus.getWisBonus();
        vitBonus = statBonus.getVitBonus();
        lifeBonus = statBonus.getLifeBonus();
        manaBonus = statBonus.getManaBonus();

    }

    public int getAttBonus() {
        return attBonus;
    }

    public int getDefBonus() {
        return defBonus;
    }

    public int getSpdBonus() {
        return spdBonus;
    }

    public int getDexBonus() {
        return dexBonus;
    }

    public int getVitBonus() {
        return vitBonus;
    }

    public int getWisBonus() {
        return wisBonus;
    }

    public int getLifeBonus() {
        return lifeBonus;
    }

    public int getManaBonus() {
        return manaBonus;
    }
}

File: app/src/main/java/com/waynebloom/rotmgdpscalculator/StatBonus.kt
package com.waynebloom.rotmgdpscalculator

internal class StatBonus {
    var attBonus = 0
    var defBonus = 0
    var spdBonus = 0
    var dexBonus = 0
    var vitBonus = 0
    var wisBonus = 0
    var lifeBonus = 0
    var manaBonus = 0

    constructor() {}
    constructor(
        attBonus: Int,
        defBonus: Int,
        spdBonus: Int,
        dexBonus: Int,
        wisBonus: Int,
        vitBonus: Int,
        lifeBonus: Int,
        manaBonus: Int
    ) {
        this.attBonus = attBonus
        this.defBonus = defBonus
        this.spdBonus = spdBonus
        this.dexBonus = dexBonus
        this.wisBonus = wisBonus
        this.vitBonus = vitBonus
        this.lifeBonus = lifeBonus
        this.manaBonus = manaBonus
    }

    fun addBonus(statBonus: StatBonus?) {
        attBonus = attBonus + statBonus!!.attBonus
        defBonus = defBonus + statBonus.defBonus
        spdBonus = spdBonus + statBonus.spdBonus
        dexBonus = dexBonus + statBonus.dexBonus
        wisBonus = wisBonus + statBonus.wisBonus
        vitBonus = vitBonus + statBonus.vitBonus
        lifeBonus = lifeBonus + statBonus.lifeBonus
        manaBonus = manaBonus + statBonus.manaBonus
    }

    fun subtractBonus(statBonus: StatBonus) {
        attBonus = attBonus - statBonus.attBonus
        defBonus = defBonus - statBonus.defBonus
        spdBonus = spdBonus - statBonus.spdBonus
        dexBonus = dexBonus - statBonus.dexBonus
        wisBonus = wisBonus - statBonus.wisBonus
        vitBonus = vitBonus - statBonus.vitBonus
        lifeBonus = lifeBonus - statBonus.lifeBonus
        manaBonus = manaBonus - statBonus.manaBonus
    }

    fun replaceBonus(statBonus: StatBonus) {
        attBonus = statBonus.attBonus
        defBonus = statBonus.defBonus
        spdBonus = statBonus.spdBonus
        dexBonus = statBonus.dexBonus
        wisBonus = statBonus.wisBonus
        vitBonus = statBonus.vitBonus
        lifeBonus = statBonus.lifeBonus
        manaBonus = statBonus.manaBonus
    }
}